admin-node-setup.sh

```diff
@@ -49,15 +49,61 @@ cp $tmp_dir/*.yaml $kube_dir/
 
 rm -rf $tmp_dir
 
-# First time setup of user-configuration for salt-master
+# Create CaaSP config dir
 if [ ! -d "/etc/caasp" ]; then
     mkdir /etc/caasp
 fi
 
+# First time setup of user-configuration for salt-master
 if [ ! -f "/etc/caasp/salt-master-custom.conf" ]; then
     echo "# Custom Configurations for Salt-Master" > /etc/caasp/salt-master-custom.conf
 fi
 
+# Migrate haproxy config path post path change
+if [[ ! -f "/etc/caasp/haproxy/haproxy.cfg" && -f "/etc/haproxy/haproxy.cfg" ]]; then
+    if [ ! -d "/etc/caasp/haproxy" ]; then
+        mkdir /etc/caasp/haproxy
+    fi
+
+    mv /etc/haproxy/haproxy.cfg /etc/caasp/haproxy/haproxy.cfg
+
+    # Add the Velum and Velum-API services to HAproxy
+    cat << EOF >> /etc/caasp/haproxy/haproxy.cfg
+
+listen velum
+        bind 0.0.0.0:80
+        bind 0.0.0.0:443 ssl crt /etc/pki/velum.pem ca-file /etc/pki/ca.crt
+        mode http
+        acl path_autoyast path_reg ^/autoyast$
+        option forwardfor
+        http-request set-header X-Forwarded-Proto https
+        redirect scheme https code 302 if !{ ssl_fc } !path_autoyast
+        default-server inter 10s fall 3
+        balance roundrobin
+        server velum unix@/var/run/puma/dashboard.sock
+
+listen velum-api
+        bind 127.0.0.1:443 ssl crt /etc/pki/velum.pem ca-file /etc/pki/ca.crt
+        mode http
+        option forwardfor
+        http-request set-header X-Forwarded-Proto https
+        default-server inter 10s fall 3
+        balance roundrobin
+        server velum unix@/var/run/puma/api.sock
+EOF
+fi
```

Comment:
Maybe I'm misunderstanding, but do we really need to move the configuration from the old location to the new? When the package with the new location is installed, shouldn't this config be installed as part of the package installation? 

https://github.com/kubic-project/caasp-container-manifests/commit/e8ace8fe824383bc0118520cf20de07d0358d564 and https://github.com/kubic-project/caasp-container-manifests/commit/d707d7dda42d18cb1e4587a9b63a61a6796c2250

Reply:
Yes, we need to move it. If we revert to the default configuration,  then we end up removing the Dex and Kube-API proxies. These will come back once the user eventually upgrades all nodes, but, we should keep it working for the period where they haven't done that.